gui/src/main/java/kcl/teamIndexZero/traffic/gui/GUI_Primitives.java

```diff
@@ -0,0 +1,138 @@
+package kcl.teamIndexZero.traffic.gui;
+
+import com.sun.org.apache.xml.internal.resolver.helpers.PublicId;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.geom.AffineTransform;
+import java.awt.geom.QuadCurve2D;
+import java.io.*;
+
+/**
+ * Created by kumar awijeet on 2/24/2016. thanks for Working!!
+ */
+public class GUI_Primitives {
+    public void drawLine(int x1, int y1, int x2, int y2, Graphics g) {
+        Graphics2D g2 = (Graphics2D) g;
+        g.drawLine(x1, y1, x2, y2);
+    }
+
+    public void drawCurve(int x1, int y1, int x2, int y2, int ctrlx, int ctrly, Graphics g) {
+        Graphics2D g2 = (Graphics2D) g;
+        QuadCurve2D q = new QuadCurve2D.Float();
+        q.setCurve(x1, y1, ctrlx, ctrly, x2, y2);
+        g2.draw(q);
+    }
+
+    public void drawSingleRoad(int x, int y, int x1, int y1, int x2, int y2, int x3, int y3, Graphics g) {
+        drawLine(x, y, x1, y1, g);
+        drawLine(x2, y2, x3, y3, g);
+    }
+    public void drawTruck(int x3, int y3, double angleToXAxis, Graphics g)
+    {
+        //files taken from http://hdimagelib.com/trailer+truck+top+view
+        g.setColor(Color.black);
+        AffineTransform c;
+        Graphics2D h = (Graphics2D) g;
+        c = h.getTransform();
+        AffineTransform at = new AffineTransform();
+        at.rotate(angleToXAxis,x3,y3);
+        h.setTransform(at);
+        Image img = null;
+        InputStream i = null;
+        try {
+            i = new BufferedInputStream(new FileInputStream("gui/src/main/resources/sprites/8460_st0640_117.jpg"));
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        try {
+            assert i != null;
+            img = ImageIO.read(i);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        h.drawImage(img,x3-18,y3-18,40,40,null);
+        h.setTransform(c);
+    }
+    public void drawSmallCar(int x3, int y3, double angleToXAxis, Graphics g)
```

Comment:
If I convert these into one single one, I will need one extra parameter which will make us choose the car. Like maybe have number 1 for small car and 2 for truck... this will mean selecting the numbers from your side, thus, I think it's better to have different functions, and call them individually. Even your method is ok, can implement it with a simple switch case for the image read, and let the remaining process be same.                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone on the EE network.                                                                                                                                                                                                                From: Alexander CherednichenkoSent: Sunday 20 March 2016 23:18To: teamIndexZero/index_zero_trafficsystemReply To: teamIndexZero/index_zero_trafficsystemCc: kumar awijeetSubject: Re: [index_zero_trafficsystem] Gui graphics kumar awijeet (#9)In gui/src/main/java/kcl/teamIndexZero/traffic/gui/GUI_Primitives.java:
> +        InputStream i = null;
> +        try {
> +            i = new BufferedInputStream(new FileInputStream("gui/src/main/resources/sprites/8460_st0640_117.jpg"));
> +        } catch (FileNotFoundException e) {
> +            e.printStackTrace();
> +        }
> +        try {
> +            assert i != null;
> +            img = ImageIO.read(i);
> +        } catch (IOException e) {
> +            e.printStackTrace();
> +        }
> +        h.drawImage(img,x3-18,y3-18,40,40,null);
> +        h.setTransform(c);
> +    }
> +    public void drawSmallCar(int x3, int y3, double angleToXAxis, Graphics g)

This code seems to be a full copy-and-paste of the previuos method. It is much better to make a single method with a parameter - or, if you want, two methods which will pass the parameter. Something like: 

// note it is private - used only inside.
private void drawCar(Graphics, ..., String resourceName) {
//    ... do actual drawing code, loading stuff from resourceName
} 

public void drawTruck(Graphics graphics, ... ) {
   drawCar(graphics, "/resources/truck_sprite") ;
}

public void drawSmallCar(Graphics graphics, ... ) {
   drawCar(graphics, "/resources/small_sprite") ;
}

https://en.wikipedia.org/wiki/Don%27t_repeat_yourself that's a starter point to get to know why repeating code is bad.

—You are receiving this because you authored the thread.Reply to this email directly or view it on GitHub